### PR TITLE
Remove ReactWindowsCore projects

### PR DESCRIPTION
**Pull request by JunielKatarn**
Source folder and projects ReactWindowsCore have become redundant, and only increase project structure complexity:
- A shared (VCXITEMS) project already exists and is used for its current purpose: `Shared\Shared.vcxitems`.
- Since Desktop and Universal builds take place in different solutions, there is currently no shared static library that would reduce build times.
- Since Desktop and Universal have different settings for Hermes and V8, two separate static libs had been created: `ReactWindowsCore.vcxproj` and `ReactWindowsCore-Desktop.vcxproj`. \
These were directly consumed ONLY by their corresponding Desktop and Universal projects, thus adding no build benefit over a shared items project (i.e. `Shared.vcxitems`).

This change simplifies the solution structure moving the shared source files into their intended project.

*Note*: This change seems rather large, but it's only because sources are moved from the `ReactWindowsCore` subdirectory to `Shared`.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5012)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj [hunk 1]
```diff
@@ -314,6 +313,7 @@
     </ClInclude>
     <ClInclude Include="LifecycleState.h" />
     <ClInclude Include="Modules\AppStateModule.h" />
+    <ClInclude Include="Modules\AsyncStorageModuleWin32.h" />
```
Comment (vmoroz) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:316]: >AsyncStorageModuleWin32.h [](start = 32, length = 25)

Having win32 specific module in a UWP project seems to be wrong.

File: vnext/Desktop/ABI/NativeTracing.idl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-namespace facebook.react {
+namespace Microsoft.React {
```
Comment (vmoroz) [inline vnext/Desktop/ABI/NativeTracing.idl:1]: >Microsoft.React [](start = 10, length = 15)

If we change the namespace, then it would be better to use the Microsoft.ReactNative to keep it in sync with the Microsoft.ReactNative.dll

Comment (asklar) [reply vnext/Desktop/ABI/NativeTracing.idl:1]: Microsoft.ReactNative.Desktop ?

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:316]: This file should probably be delted, but I didn't want to take that step in this change.
@htpiv mentioned the base React API (AsyncStorageModule) was dropped in 0.62.

File: vnext/Desktop/ABI/NativeTracing.idl (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Desktop/ABI/NativeTracing.idl:1]: I believe there was a potential conflict issue if the same namespace was used. Is that not accurate?

@asklar We already have several types under namespace `Microsoft::React` before `Microsoft::ReactNative` was introduced.
Adding a third namespace would make things even more confusing.

I'd like to follow up offline for guidance regarding the definitive namespaces to use.
For now, it's very clear that using `facebook::react` for non-Facebook sources is wrong.

Comment (vmoroz) [reply vnext/Desktop/ABI/NativeTracing.idl:1]: We are working towards common "core dll" to be shared between UWP and Win32. It would be better if we use the same namespace to help us there. We use the Microsoft.ReactNative namespace today. It may be changed in future, but for now it would be nice to be consistent.

---
In reply to: [430837209](https://github.com/microsoft/react-native-windows/pull/5012#discussion_r430837209) [](ancestors = 430837209)

File: vnext/Shared/WinRTWebSocketResource.cpp [hunk 3]
```diff
@@ -8,6 +8,7 @@
 #include <Utilities.h>
 
 // Windows API
+#include <winrt/Windows.Foundation.Collections.h>
```
Comment (asklar) [inline vnext/Shared/WinRTWebSocketResource.cpp:11]: why is this change needed?

File: vnext/Shared/Shared.vcxitems [hunk 4]
```diff
@@ -14,13 +15,97 @@
...
+    <ClInclude Include="$(MSBuildThisFileDirectory)tracing\fbsystrace.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)TurboModuleManager.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)TurboModuleRegistry.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Utils.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)V8JSIRuntimeHolder.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)WebSocketJSExecutorFactory.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)WinRTWebSocketResource.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="$(MSBuildThisFileDirectory)etw\build.bat" />
```
Comment (asklar) [inline vnext/Shared/Shared.vcxitems:106]: who is using this? are we plopping the ETW stuff into production binary today? 
We might want to make this a build task

File: vnext/Shared/Shared.vcxitems [hunk 5]
```diff
@@ -14,13 +15,97 @@
...
+    <ClInclude Include="$(MSBuildThisFileDirectory)Pch\pch.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)ShadowNode.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)ShadowNodeRegistry.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)targetver.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Tracing.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)tracing\fbsystrace.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)TurboModuleManager.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)TurboModuleRegistry.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Utils.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)V8JSIRuntimeHolder.h" />
```
Comment (asklar) [inline vnext/Shared/Shared.vcxitems:101]: should we mark it as excluded from build even if it's a header?

Comment (asklar) [reply vnext/Shared/Shared.vcxitems:101]: same with the Hermes one

File: vnext/Shared/Shared.vcxitems [hunk 6]
```diff
@@ -4,6 +4,7 @@
     <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
     <HasSharedItems>true</HasSharedItems>
     <ItemsProjectGuid>{2049dbe9-8d13-42c9-ae4b-413ae38fffd0}</ItemsProjectGuid>
+    <ItemsProjectName>Microsoft.React.Shared</ItemsProjectName>
```
Comment (asklar) [inline vnext/Shared/Shared.vcxitems:7]: we should reuse existing terminology - we don't use React anywhere today, so let's call this `Microsoft.ReactNative.Shared` instead

File: vnext/ReactWindows-Desktop.sln [hunk 7]
```diff
@@ -81,7 +71,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "DynamicLibrary", "DynamicLi
 		PropertySheets\DynamicLibrary\x86.props = PropertySheets\DynamicLibrary\x86.props
 	EndProjectSection
 EndProject
-Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Shared", "Shared\Shared.vcxitems", "{2049DBE9-8D13-42C9-AE4B-413AE38FFFD0}"
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Microsoft.React.Shared", "Shared\Shared.vcxitems", "{2049DBE9-8D13-42C9-AE4B-413AE38FFFD0}"
```
Comment (asklar) [inline vnext/ReactWindows-Desktop.sln:74]: `Microsoft.ReactNative.Shared` here and elsewhere (rename Shared.vcxitems too?)

File: vnext/ReactUWP/Base/CoreUIManagers.cpp [hunk 8]
```diff
@@ -6,7 +6,7 @@
 #include <Modules/NativeUIManager.h>
 #include <ReactUWP/IReactInstance.h>
 #include <ReactUWP/ViewManagerProvider.h>
-#include <ReactWindowsCore/IUIManager.h>
+#include <Shared/IUIManager.h>
```
Comment (asklar) [inline vnext/ReactUWP/Base/CoreUIManagers.cpp:9]: is the Shared/ prefix here and elsewhere, needed? If Shared is in the include path this should not be necessary

File: vnext/Shared/Shared.vcxitems (hunk 4 above)
Comment (JunielKatarn) [reply vnext/Shared/Shared.vcxitems:106]: That's a great question.
I only moved the files and added them to the solution for viewing.
I don't think they're being used by anyone.
Also, there seems to be an actual DLL binary in those sources.

File: vnext/PropertySheets/ReactDirectories.props [hunk 9]
```diff
@@ -15,14 +15,17 @@
 
     <ReactNativePackageDir Condition="'$(ReactNativePackageDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'node_modules\react-native\package.json'))\node_modules\react-native\</ReactNativePackageDir>
 
+    <RootDir>$(ReactNativeWindowsDir)</RootDir>
```
Comment (asklar) [inline vnext/PropertySheets/ReactDirectories.props:18]: we should limit the amount of special sauce needed. Can we just reuse the standard OutDir and IntDir instead?

File: vnext/Microsoft.ReactNative/Modules/AsyncStorageModuleWin32.cpp [hunk 10]
```diff
@@ -242,7 +242,7 @@ AsyncStorageModuleWin32::~AsyncStorageModuleWin32() {
     // condition_variable for the async task to acknowledge cancellation by
     // nulling out m_action. Once m_action is null, it is safe to proceed
     // wth closing the DB connection
-    winrt::slim_shared_lock_guard guard{m_lock};
+    winrt::slim_lock_guard guard{m_lock};
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/Modules/AsyncStorageModuleWin32.cpp:245]: this seems unrelated? is this intentional/why?

File: vnext/Shared/Shared.vcxitems (hunk 5 above)
Comment (JunielKatarn) [reply vnext/Shared/Shared.vcxitems:101]: I don't think there is a need for that.
Just ignoring it also reduces the "MsBuild hacking" around custom variables (such as `USE_V8`).

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters [hunk 11]
```diff
@@ -670,6 +675,7 @@
     <ClInclude Include="Views\ReactRootControl.h">
       <Filter>Views</Filter>
     </ClInclude>
+    <ClInclude Include="Modules\AsyncStorageModuleWin32.h">
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters:678]: this and the win32 cpp should not be in the main project if it's win32 only.

File: vnext/Shared/Shared.vcxitems (hunk 6 above)
Comment (JunielKatarn) [reply vnext/Shared/Shared.vcxitems:7]: Done (reflected on next push).

File: vnext/ReactWindows-Desktop.sln (hunk 7 above)
Comment (JunielKatarn) [reply vnext/ReactWindows-Desktop.sln:74]: There's no need to rename the VCXITEMS file, nor directory, as long as the `ProjectName` or `ItemsProjectName` property is set correctly.

File: vnext/ReactUWP/Base/CoreUIManagers.cpp (hunk 8 above)
Comment (JunielKatarn) [reply vnext/ReactUWP/Base/CoreUIManagers.cpp:9]: I'll give it a try.

File: vnext/ReactWindowsCore/packages.ReactWindowsCore-Desktop.config [hunk 12]
```diff
@@ -1,6 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<packages>
-  <package id="boost" version="1.68.0.0" targetFramework="native" />
-  <package id="ReactNative.Hermes.Windows" version="0.1.6" targetFramework="native" Condition="$(USE_HERMES)' == 'true'" />
```
Comment (asklar) [inline vnext/ReactWindowsCore/packages.ReactWindowsCore-Desktop.config:4]: I'm not sure how you're making sure that the desktop project installs the V8/Hermes packages if there isn't a similar packages.config file for a desktop-specific static library? Can you clarify what is the mechanism that enables this?
Also, we should make sure that creating a new app and building it, does not cause us to download/include V8/Hermes packages.

File: vnext/PropertySheets/ReactDirectories.props (hunk 9 above)
Comment (JunielKatarn) [reply vnext/PropertySheets/ReactDirectories.props:18]: This gives flexibility for custom builds.
MSVC doesn't really set up an output artifact structure that consolidates `Platform`, `Configuration`, and project path.
These additions allow for placing either the intermediate outputs, final outputs, or both in a different directory with very compact expressivity.

In short, they're just transparent build variables that won't affect the average case.

File: vnext/Microsoft.ReactNative/Modules/AsyncStorageModuleWin32.cpp (hunk 10 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Modules/AsyncStorageModuleWin32.cpp:245]: `slim_shared_lock_guard` doesn't seem available in C++/WinRT.
I'll triplecheck.

Note, these sources will be removed soon, due to RN 0.62 dropping of AsyncStorageModule.

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters (hunk 11 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters:678]: It can't be moved to the Desktop projects, because it demands Win10 or later.
It also doesn't justify a full static library project, IMHO.

Should we go ahead and delete it, since it's not relevant for RN 0.62?

File: vnext/Shared/WinRTWebSocketResource.cpp (hunk 3 above)
Comment (JunielKatarn) [reply vnext/Shared/WinRTWebSocketResource.cpp:11]: After the move to the shared project, it fails due to usage of winrt `IVector`.
```
D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(97,58): error C3779: 'winrt::impl::consume_Windows_Foundation_Collections_IVector<winrt::Windows::Foundation::Collections::IVector<winrt::Windows::Security::Cryptography::Certificates::ChainValidatio
       nResult>,T>::Append': a function that returns 'auto' cannot be used before it is defined [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
                 with
                 [
                     T=winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult
                 ]
       D:\Extra\react\windows\fixcommon\vnext\build\x64\Debug\React.Windows.Desktop\Generated Files\winrt\impl\Windows.Foundation.Collections.0.h(432): message : see declaration of 'winrt::impl::consume_Windows_Foundation_Collections_IVector<winrt::Windows::Foundation::Collectio
       ns::IVector<winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult>,T>::Append' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
                 with
                 [
                     T=winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult
                 ]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(127,41): error C3312: no callable 'await_resume' function found for type 'winrt::Windows::Foundation::IAsyncAction' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(171,39): error C3312: no callable 'await_resume' function found for type 'winrt::Windows::Storage::Streams::DataWriterStoreOperation' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Des
       ktop.vcxproj]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(239,39): error C3312: no callable 'await_resume' function found for type 'winrt::Windows::Storage::Streams::DataWriterStoreOperation' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Des
       ktop.vcxproj]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(373,23): error C3779: 'winrt::impl::consume_Windows_Foundation_Collections_IVector<winrt::Windows::Foundation::Collections::IVector<winrt::hstring>,T>::Append': a function that returns 'auto' cannot
       be used before it is defined [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
```

Actually, what puzzles me is why it was NOT failing to compile before.

File: vnext/Shared/Shared.vcxitems (hunk 4 above)
Comment (asklar) [reply vnext/Shared/Shared.vcxitems:106]: looks like it was added in this PR: #3745
There is mention of tools to process the systrace (through JS HUD?) and ETW events (WPA) but I don't see any docs on how to do so.
@mganandraj could you please write something down into an md file? Also, does this DLL need to be checked in? who is using it?

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters (hunk 11 above)
Comment (asklar) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters:678]: if it's not needed, yes please delete :)

File: vnext/Microsoft.ReactNative/Modules/AsyncStorageModuleWin32.cpp (hunk 10 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Modules/AsyncStorageModuleWin32.cpp:245]: Reason: `slim_shared_lock_guard` is available in `C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt\base.h`, but not in the CppWinRT NuGet sources:
https://github.com/microsoft/cppwinrt/search?q=slim_shared_lock_guard&unscoped_q=slim_shared_lock_guard

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters (hunk 11 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj.filters:678]: My pleasure!

File: vnext/Shared/WinRTWebSocketResource.cpp (hunk 3 above)
Comment (asklar) [reply vnext/Shared/WinRTWebSocketResource.cpp:11]: @JunielKatarn The reason this is needed now is because you deleted the ReactWindowsCore/pch folder (there doesn't seem to be a Shared/pch folder anymore). The PCH used to bring along a bunch of WinRT headers, including the one you're having to include manually now.

I wonder if this is fine or if it will make it harder for WinUI 3... I suppose if the CI passes the WinUI 3 task then this should be ok.

Comment (JunielKatarn) [reply vnext/Shared/WinRTWebSocketResource.cpp:11]: CI does pass WinUI 3.

Please note, there is a `Shared\Pch` directory:
```
 ls .\Shared\Pch\
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           5/26/2020  3:04 PM            118 pch.cpp
-a---           5/26/2020  3:04 PM            359 pch.h
```

However, being a shared project and not a concrete one (i.e. static lib), the PCH might be obtained form elsewhere.
In short, any critical includes should be added to either the Desktop VCXPROJ, MS.ReactNative VCXPROJ, or both.

This doesn't seem to be the case. The winrt Collections header is not being widely used.

File: vnext/ReactWindowsCore/packages.ReactWindowsCore-Desktop.config (hunk 12 above)
Comment (JunielKatarn) [reply vnext/ReactWindowsCore/packages.ReactWindowsCore-Desktop.config:4]: That's correct. I was missing adding it to the Desktop project.
Solved.

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj]: Removed.

**Comment by JunielKatarn** [comment]
Issues addressed. Awaiting final approval.

File: vnext/ReactUWP/Base/CoreNativeModules.cpp [hunk 13]
```diff
@@ -116,7 +115,7 @@ std::vector<facebook::react::NativeModuleDescription> GetCoreModules(
         if (HasPackageIdentity()) {
```
Comment (asklar) [inline vnext/ReactUWP/Base/CoreNativeModules.cpp:115]: is this right? do we need to be packaged for the storage APIs to work? this seems in opposition to not needing the win32 module. I'm hoping we can always use the WinRT storage APIs whether we're packaged or not

Comment (JunielKatarn) [reply vnext/ReactUWP/Base/CoreNativeModules.cpp:115]: I’d have to check with the author of this method.

Can we leave this investigation for a related subsequent PR?

File: vnext/ReactUWP/Base/CoreNativeModules.cpp [hunk 14]
```diff
@@ -117,7 +115,7 @@ std::vector<facebook::react::NativeModuleDescription> GetCoreModules(
...
           return std::make_unique<facebook::react::AsyncStorageModule>(L"asyncStorage");
         } else {
-          return std::make_unique<facebook::react::AsyncStorageModuleWin32>();
+          return nullptr;
```
Comment (vmoroz) [inline vnext/ReactUWP/Base/CoreNativeModules.cpp:118]: >return nullptr; [](start = 10, length = 15)

We cannot do it. Adding a nullptr for the module will crash all users of XAML islands.

File: vnext/Shared/Shared.vcxitems [hunk 15]
```diff
@@ -14,13 +15,97 @@
...
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\ExceptionsManagerModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\I18nModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\PlatformConstantsModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\SourceCodeModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\StatusBarManagerModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\UIManagerModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Modules\WebSocketModule.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)NativeModuleProvider.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)OInstance.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)Pch\pch.h" />
```
Comment (vmoroz) [inline vnext/Shared/Shared.vcxitems:92]: >  <ClInclude Include="$(MSBuildThisFileDirectory)Pch\pch.h" /> [](start = 2, length = 62)

Shared project must not have a pch to avoid interfering with project's pch that includes it.

File: vnext/ReactUWP/Base/CoreNativeModules.cpp (hunk 13 above)
Comment (acoates-ms) [reply vnext/ReactUWP/Base/CoreNativeModules.cpp:115]: I suspect you need to be packaged to get an identity for the storage APIs to know where to store the info.

I'm not too concerned with the functionality here, as AsyncStorage has already been deprecated, and will probably be removed in a release or two.

File: vnext/Shared/Shared.vcxitems (hunk 4 above)
Comment (mganandraj) [reply vnext/Shared/Shared.vcxitems:106]: Hi .. The resource dll has the schema for the ETW events payload.. Developers can register this resource file against the ETW provider so that teh tools like WPA can parse the ETW events payload.. There is a batch file checked in along with the resource dll to build and register it on a developer machine...  

The resource dll is not supposed to be shipped to the user machine .. It is checked in to make the ETW analysis easier even on a machine with only Windows Perf Toolkit installed but not the build toolchains ..

**Comment by vmoroz** [comment]
Removing extra projects is a good change, but it also includes a number of code changes that raise questions. Would it be possible to do them one-by-one instead? For example:
- We should use Microsoft.ReactNative for our public ABI namespace, not Microsoft.React. It can be a subject of another PR.
- No need to change the async storage code. It can be kept as it was before for now and if needed changed later.
- Since some files were moved and changed, it is difficult to know what was changed. It would be nice to avoid changing files that are moved and change them after the move.

File: vnext/Desktop/ABI/NativeTracing.idl (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Desktop/ABI/NativeTracing.idl:1]: As discussed offline, I'll scope down the change and remove any namespace renamings.

File: vnext/Shared/Shared.vcxitems (hunk 4 above)
Comment (JunielKatarn) [reply vnext/Shared/Shared.vcxitems:106]: @mganandraj  Can't we consume that DLL as a NuGet dependency?

We shoud NOT push binaries into a Git repository.

Comment (mganandraj) [reply vnext/Shared/Shared.vcxitems:106]: I don't have any strong opinion here.. This is not a code dll, and it rarely changes and hence there is not overhead of version tracking .. Unfortunately, I'm quite busy with some other tasks for this quarter hence won't be able to get to any cleanups here..

File: vnext/Desktop/ABI/MemoryTracker.idl [hunk 16]
```diff
@@ -3,7 +3,7 @@ import "ABI/MessageQueue.idl";
 // ABI_REVIEW:
 // Given this is a new API without usage legacy, should we use the chance to introduce a different
 // namespace, e.g. "Microsoft.React"?
-namespace facebook.react {
+namespace Microsoft.React {
```
Comment (aeulitz) [inline vnext/Desktop/ABI/MemoryTracker.idl:6]: >Microsoft.React  [](start = 10, length = 16)

While a meaningful change, is it a necessary part of the project restructuring?

File: vnext/Desktop.ABITests/ActivationFactory.cpp [hunk 17]
```diff
@@ -26,7 +26,7 @@ int32_t __stdcall WINRT_RoGetActivationFactory(void *classId, guid const &iid, v
   std::wstring_view const name{*reinterpret_cast<winrt::hstring *>(&classId)};
   HMODULE library{nullptr};
 
-  if (starts_with(name, L"facebook.react.") || starts_with(name, L"Microsoft.React")) {
```
Comment (aeulitz) [inline vnext/Desktop.ABITests/ActivationFactory.cpp:29]: >  if (starts_with(name, L"facebook.react.") || starts_with(name, L"Microsoft.React"))  [](start = 0, length = 86)

My latest change has removed this file.

File: change/react-native-windows-2020-05-26-19-03-03-fixcommon.json [hunk 18]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "prerelease",
+  "comment": "Remove references to ReactWindowsCore projects",
```
Comment (aeulitz) [inline change/react-native-windows-2020-05-26-19-03-03-fixcommon.json:3]: >Remove references [](start = 14, length = 17)

Nit: Looks like it's not just the references that get removed :)

**Comment by aeulitz** [comment]
> class AsyncStorageManager {

For my education: Can this class be used for both Win32 and UWP? Is this a candidate for a (future) DLL of native modules? 

---
Refers to: vnext/Shared/AsyncStorage/AsyncStorageManager.h:17 in b16236d. [](commit_id = b16236d5f492ea0bd0829791dff68c988df5b03d, deletion_comment = False)

File: vnext/ReactUWP/Base/CoreNativeModules.cpp (hunk 13 above)
Comment (JunielKatarn) [reply vnext/ReactUWP/Base/CoreNativeModules.cpp:115]: I have worked around this. Won't remove that Win32 module, nor move it into the main project.

File: vnext/ReactUWP/Base/CoreNativeModules.cpp (hunk 14 above)
Comment (JunielKatarn) [reply vnext/ReactUWP/Base/CoreNativeModules.cpp:118]: Worked around it. Reverting.

File: vnext/Desktop/ABI/MemoryTracker.idl (hunk 16 above)
Comment (JunielKatarn) [reply vnext/Desktop/ABI/MemoryTracker.idl:6]: It's not. Already agreed to revert.

**Comment by JunielKatarn** [comment]
> > class AsyncStorageManager {
> 
> For my education: Can this class be used for both Win32 and UWP? Is this a candidate for a (future) DLL of native modules?
> 
> Refers to: vnext/Shared/AsyncStorage/AsyncStorageManager.h:17 in [b16236d](https://github.com/microsoft/react-native-windows/commit/b16236d5f492ea0bd0829791dff68c988df5b03d). [](commit_id = [b16236d](https://github.com/microsoft/react-native-windows/commit/b16236d5f492ea0bd0829791dff68c988df5b03d), deletion_comment = False)

Not sure, unfortunately.

File: change/react-native-windows-2020-05-26-19-03-03-fixcommon.json (hunk 18 above)
Comment (JunielKatarn) [reply change/react-native-windows-2020-05-26-19-03-03-fixcommon.json:3]: Ended up scaling this down. The sources will stay in that folder for the time being, but the projects are indeed removed.

**Comment by JunielKatarn** [comment]
@vmoroz I have scaled down the size of this change:
- Removed ONLY the ReactWindowsCore project, but left the sources in place so the change is not deemed as "too big". Will revisit moving the sources after we start working on 0.63.
- Reverted the Desktop root namespace renaming. Will submit it on a separate change.

C.C. @asklar

File: vnext/ReactWindowsCore/Modules/AsyncStorageModuleWin32.cpp [hunk 19]
```diff
@@ -242,7 +242,7 @@ AsyncStorageModuleWin32::~AsyncStorageModuleWin32() {
     // condition_variable for the async task to acknowledge cancellation by
     // nulling out m_action. Once m_action is null, it is safe to proceed
     // wth closing the DB connection
-    winrt::slim_shared_lock_guard guard{m_lock};
```
Comment (vmoroz) [inline vnext/ReactWindowsCore/Modules/AsyncStorageModuleWin32.cpp:245]: Can we keep this code unchanged? I understand that the shared lock allows multiple readers to access data.

File: vnext/ReactCommon/ReactCommon.vcxproj [hunk 20]
```diff
@@ -189,6 +189,8 @@
     <!-- Forced include only necessary until we get https://github.com/facebook/react-native/pull/28023  back from upstream. -->
     <ClCompile Include="$(ReactNativeDir)\ReactCommon\turbomodule\core\TurboModuleBinding.cpp" ForcedIncludeFiles="stdexcept" />
     <ClCompile Include="$(ReactNativeDir)\ReactCommon\turbomodule\core\TurboModuleUtils.cpp" />
+    <CLCompile Include="$(ReactNativeDir)\ReactCommon\turbomodule\samples\NativeSampleTurboCxxModuleSpecJSI.cpp" />
```
Comment (vmoroz) [inline vnext/ReactCommon/ReactCommon.vcxproj:192]: Can we avoid this change? We have a different implementation for TurboModules.

File: vnext/PropertySheets/ReactDirectories.props (hunk 9 above)
Comment (vmoroz) [inline vnext/PropertySheets/ReactDirectories.props:18]: It is better to have names started with React to avoid any conflict with names that user code or VS code has.

File: vnext/Desktop/ABI/MessageQueueShim.h [hunk 21]
```diff
@@ -4,7 +4,7 @@
 #include <winrt/facebook.react.h>
 
 namespace facebook::react {
-class MessageQueueShim : public MessageQueueThread {
+class MessageQueueShim : public facebook::react::MessageQueueThread {
```
Comment (vmoroz) [inline vnext/Desktop/ABI/MessageQueueShim.h:7]: revert?

File: docs/project-structure.md [hunk 22]
```diff
@@ -45,7 +40,7 @@ Holds sources common to both Windows variants, that require different build conf
 *Static Library*<br/>
 React Native core, cross-platform C++ types and interfaces.<br/>
 Sources provided as part of the `react-native` Node dependency. Not part of this repository.<br/>
-See https://github.com/facebook/react-native/tree/v0.59.9/ReactCommon.
+See https://github.com/facebook/react-native/tree/v0.60.0/ReactCommon.
```
Comment (vmoroz) [inline docs/project-structure.md:43]: We are releasing version 0.62

Comment (JunielKatarn) [reply docs/project-structure.md:43]: True. Typo.

File: vnext/ReactWindowsCore/Modules/AsyncStorageModuleWin32.cpp (hunk 19 above)
Comment (JunielKatarn) [reply vnext/ReactWindowsCore/Modules/AsyncStorageModuleWin32.cpp:245]: I haven't been able to. The header is not available in Universal projects, and the SQLite API is not available in Win8.1.

Note, this module is deprecated starting 0.62.

File: vnext/ReactCommon/ReactCommon.vcxproj (hunk 20 above)
Comment (JunielKatarn) [reply vnext/ReactCommon/ReactCommon.vcxproj:192]: I think I encountered a build issue, but I'll try.

File: vnext/PropertySheets/ReactDirectories.props (hunk 9 above)
Comment (JunielKatarn) [reply vnext/PropertySheets/ReactDirectories.props:18]: I researched the already defined MSVC properties.
These don't collide with those.
It's fair to ask users using this framework to chose their own custom properties.
Moreover, it's highly unlikely that React Native developers would have an issue with MSBuild property names.

File: vnext/ReactCommon/ReactCommon.vcxproj (hunk 20 above)
Comment (JunielKatarn) [reply vnext/ReactCommon/ReactCommon.vcxproj:192]: I was unable to make the solutions build without these sources.

I'll follow up in the next PR to exclude it form any projects it could be intrusive to.

**Comment by jonthysell** [comment]
@JunielKatarn Please make a corresponding update to the docs repo for native modules setup to indicate that this project is no longer present nor necessary. If this is getting backported to 62, you'll want to make these changes now, before the website releases the 62 docs. Otherwise, you want to wait until after the website snaps to 62, and then update the docs.

**Comment by JunielKatarn** [comment]
@jonthysell certainly. I'll address this today.

**Comment by NickGerleman** [comment]
Removing the `Backport to 0.62` label, since we need to rename it as part of the deprecation in favor of `Request Backport` which was already added to this PR. Triage of this item should still happen soon.

**Comment by Khalef1** [comment]
We discussed this in triage and decided not to backport the change. It's nice to have but will also cause a breaking change to the template project
